packages/fx-core/src/common/tools.ts

```diff
@@ -948,3 +950,39 @@ export async function getSPFxToken(
   }
   return spoToken;
 }
+
+// Get some fields from project settings and will be used in telemetry
+export function getProjectSettingsForCommonTelemetry(rootPath: string | undefined): any {
+  if (!rootPath) {
+    return undefined;
+  }
+
+  try {
+    const projectSettingsPath = path.join(
+      rootPath,
+      `.${ConfigFolderName}`,
+      InputConfigsFolderName,
+      ProjectSettingsFileName
+    );
+
+    if (!projectSettingsPath || !fs.pathExistsSync(projectSettingsPath)) {
+      return undefined;
+    }
+
+    const projectSettings = fs.readJsonSync(projectSettingsPath);
```

Comment:
Is there a common method for loading project settings? Reading the raw project setting file may be not a good practice.